src/libstore/legacy-ssh-store.cc

```diff
@@ -18,7 +18,12 @@ struct LegacySSHStoreConfig : virtual StoreConfig
     const Setting<Path> sshKey{(StoreConfig*) this, "", "ssh-key", "path to an SSH private key"};
     const Setting<std::string> sshPublicHostKey{(StoreConfig*) this, "", "base64-ssh-public-host-key", "The public half of the host's SSH key"};
     const Setting<bool> compress{(StoreConfig*) this, false, "compress", "whether to compress the connection"};
-    const Setting<Path> remoteProgram{(StoreConfig*) this, "nix-store", "remote-program", "path to the nix-store executable on the remote system"};
+    const Setting<Path> remoteProgram{
+        (StoreConfig*) this,
+        getEnv("NIX_LEGACY_SSH_REMOTE_PROGRAM").value_or("nix-store"),
```

Comment:
There is a bit of a convention to prefix these variable with `_NIX_TEST_`. Alternatively, we can avoid the environment variable by appending `?remote-program=...` to the store URL.

Reply:
I switched it to `_NIX_TEST_` instead. As to `?remote-program=...` I can pursue if you want, but do note we have some fancy ssh urls e.g. doing iterative ssh with `?remote-store=` so it is a bit a non-trivial. I am thinking the env var might be better because it means no one can forget the remote program part going forward if they are adding another ssh test, especially if it is also complicated.